reconcile/openshift_resources_base.py

```diff
@@ -234,11 +248,30 @@ def lookup_graphql_query_results(query: str, **kwargs) -> list[Any]:
     return results
 
 
-def process_jinja2_template(body, vars=None, env=None, settings=None):
+@lru_cache
```

Comment:
I love this idea. Have you tried with `@cache`? Since we have already a lot of things in memory, we may want to add a few more with an unbound cache 😬

Reply:
fixed

Reply:
i did not consider an unbounded cache until now. thinking about this, i'm not too much concerned about this cache filling up uncontrolled. we don't have that many template changes that we would be hit by this. we restart reconciler way more often than we change templates.

additionally, cache eviction with lru_cache might cost performance gains. so i'm in favour of your proposal!

Reply:
changed in [7a13b0d](https://github.com/app-sre/qontract-reconcile/pull/2734/commits/7a13b0d87d07f4d90205529e58852b580805d24a)